share/picongpu/unit/radiationWindowFunction.cpp

```diff
@@ -0,0 +1,163 @@
+/* Copyright 2024 Rene Widera
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <pmacc/boost_workaround.hpp>
+
+#include <pmacc/test/PMaccFixture.hpp>
+
+// STL
+#include <pmacc/Environment.hpp>
+#include <pmacc/dimensions/DataSpace.hpp>
+#include <pmacc/lockstep.hpp>
+#include <pmacc/memory/buffers/DeviceBuffer.hpp>
+#include <pmacc/memory/buffers/HostBuffer.hpp>
+#include <pmacc/meta/conversion/MakeSeq.hpp>
+
+#include <cstdint>
+#include <iostream>
+#include <string>
+
+#include <catch2/catch_approx.hpp>
+#include <catch2/catch_test_macros.hpp>
+#include <picongpu/param/precision.param>
+#include <picongpu/plugins/radiation/windowFunctions.hpp>
+
+
+//! Helper to setup the PMacc environment
+using TestFixture = pmacc::test::PMaccFixture<TEST_DIM>;
+static TestFixture fixture;
+
+using namespace picongpu;
+using namespace pmacc;
+
+constexpr uint32_t numValues = 1024;
+constexpr uint32_t elemPerBlock = 256;
+
+/** check if floating point result is equal
+ *
+ * Allows an error of one epsilon.
+ * @return true if equal, else false
+ */
+template<typename T>
+static bool isApproxEqual(T const& a, T const& b)
+{
+    return a == Catch::Approx(b).margin(std::numeric_limits<T>::epsilon());
+}
+
+/** Test radiation window function
+ *
+ * Compares the on host and on device computed result.
+ *
+ * @tparam T_WindowFunction radiation window function
+ */
+template<typename T_WindowFunction = boost::mpl::_1>
+struct TestWindowFunction
+{
+    /** Validates the shape
+     *
+     * @param inCellPositionBuffer Buffer with positions, each value must be in rage [0.0;1.0).
```

Comment:
should be global position in [0, Lx) - using [0, 1.0) * Lx is fine - see code suddgestion below.

Reply:
I used your suggestion